[terra-list] Fixed focus issue when clicked on selectable item in draggable list item example (#4054)
cerner/terra-core: packages/terra-list/CHANGELOG.md

## Unreleased

* Fixed

>   * Fixed the issue where focus was not moving to the selected item when clicked in the draggable list item example.

Reviewer: ```suggestion
  * Fixed the issue where focus was not moving to the selectable and draggable item when clicked.
```

Author: Updated 2b1b08085c1bd7f5c18f793e30374d5bf46aa748
